internal/models/linking.go

```diff
@@ -109,14 +109,14 @@ func DetermineAccountLinking(tx *storage.Connection, config *conf.GlobalConfigur
 	var similarIdentities []*Identity
 	var similarUsers []*User
 	// look for similar identities and users based on email
-	if terr := tx.Q().Eager().Where("email ilike any (?)", verifiedEmails).All(&similarIdentities); terr != nil {
+	if terr := tx.Q().Eager().Where("email = any (?)", verifiedEmails).All(&similarIdentities); terr != nil {
 		return AccountLinkingResult{}, terr
 	}
 
 	if !strings.HasPrefix(providerName, "sso:") {
 		// there can be multiple user accounts with the same email when is_sso_user is true
 		// so we just do not consider those similar user accounts
-		if terr := tx.Q().Eager().Where("email ilike any (?) and is_sso_user is false", verifiedEmails).All(&similarUsers); terr != nil {
+		if terr := tx.Q().Eager().Where("lower(email) = any (?) and is_sso_user = false", verifiedEmails).All(&similarUsers); terr != nil {
```

Comment:
This query still takes a while when I test it on our project. I looked through the indexes and we have one on `email`, but not `lower(email)`. Adding one seemed to resolve the problem, but you may have better ideas here.

Reply:
Yeah agreed this is not going to fix the performance issues. To have queries be fast you can't use `ilike` or functions like `lower` on the column.

We need to add a (non-unique) index with `text_pattern_ops` as well @kangmingtay

Reply:
@hf @J0 will need to measure the impact of adding / updating the index on larger tables

this change improves the performance slightly without sacrificing any downtime

Reply:
Maybe we can start with this and then iterate?